solr/solrj/src/test/org/apache/solr/client/solrj/cloud/autoscaling/TestPolicy2.java

```diff
@@ -516,6 +517,16 @@ public void testInfiniteLoop() {
     System.out.println(suggestions);
   }
 
+  public void testAddTooManyPerPolicy() {
+    Map<String, Object> m = (Map<String, Object>) loadFromResource("testAddTooManyPerPolicy.json");
+    SolrCloudManager cloudManagerFromDiagnostics = createCloudManagerFromDiagnostics(m);
+    AutoScalingConfig autoScalingConfig = new AutoScalingConfig((Map<String, Object>) getObjectByPath(m, false, "diagnostics/config"));
+    SolrException exp =  expectThrows(SolrException.class, () -> PolicyHelper.getReplicaLocations("TooManyPerPolicy", autoScalingConfig, cloudManagerFromDiagnostics,
+            EMPTY_MAP, Collections.singletonList("shard1"), 1, 0, 0, null));
+    assertTrue(exp.getMessage().contains("No node can satisfy the rules"));
```

Comment:
`assertThat(exp.getMessage(), contains("..."))`

Reply:
what's the difference?

Reply:
You get a better error message when the test fails.